runtime/zeitgeist/src/xcm_config/config.rs

```diff
@@ -0,0 +1,305 @@
+    type IsReserve = MultiNativeAsset<AbsoluteReserveProvider>;
+    /// Combinations of (Location, Asset) pairs which we trust as teleporters.
+    type IsTeleporter = ();
+    /// Means of inverting a location.
+    type LocationInverter = LocationInverter<Ancestry>;
+    /// How to get a call origin from a `OriginKind` value.
+    type OriginConverter = XcmOriginToTransactDispatchOrigin;
+    /// Module that handles responses of queries.
+    type ResponseHandler = PolkadotXcm;
+    /// Module that handles subscription requests.
+    type SubscriptionService = PolkadotXcm;
+    /// The means of purchasing weight credit for XCM execution.
+    type Trader = Trader;
+    /// The means of determining an XCM message's weight.
+    // Adds UnitWeightCost per instruction plus the weight of each instruction.
+    // The total number of instructions are bounded by MaxInstructions
+    type Weigher = FixedWeightBounds<UnitWeightCost, Call, MaxInstructions>;
+    /// How to send an onward XCM message.
+    type XcmSender = XcmRouter;
+}
+
+/// Additional filters that specify whether the XCM instruction should be executed at all.
+pub type Barrier = (
+    // Execution barrier that just takes max_weight from weight_credit
+    TakeWeightCredit,
+    // Ensures that execution time is bought with BuyExecution instruction
+    AllowTopLevelPaidExecutionFrom<Everything>,
+    // Expected responses are OK.
+    AllowKnownQueryResponses<PolkadotXcm>,
+    // Subscriptions for version tracking are OK.
+    AllowSubscriptionsFrom<Everything>,
+);
+
+/// The means of purchasing weight credit for XCM execution.
+/// Every token that is accepted for XC transfers should be handled here.
+pub type Trader = (
+    // In case the asset in question is the native currency, it will charge
+    // the default base fee per second and deposits them into treasury
+    FixedRateOfFungible<ZtgPerSecond, ToTreasury>,
+    FixedRateOfFungible<ZtgPerSecondCanonical, ToTreasury>,
+    // For all other assets the base fee per second will tried to be derived
+    // through the `fee_factor` entry in the asset registry. If the asset is
+    // not present in the asset registry, the default base fee per second is used.
+    // Deposits all fees into the treasury.
+    AssetRegistryTrader<
+        FixedRateAssetRegistryTrader<FixedConversionRateProvider<AssetRegistry>>,
+        ToTreasury,
+    >,
+);
+
+pub struct ToTreasury;
+impl TakeRevenue for ToTreasury {
+    fn take_revenue(revenue: MultiAsset) {
+        use xcm_executor::traits::Convert;
+
+        if let MultiAsset { id: Concrete(location), fun: Fungible(amount) } = revenue.clone() {
+            if let Ok(asset_id) =
+                <AssetConvert as Convert<MultiLocation, CurrencyId>>::convert(location.clone())
+            {
+                let _ = AssetManager::deposit(asset_id, &ZeitgeistTreasuryAccount::get(), amount);
+            }
+        }
+    }
+}
```

Comment:
`take_revenue` is invoked when the fees for the XCM execution have been withdrawn. It tries to convert the `MultiLocation` to a `CurrencyId`. Only if that is possible, fees are taken into the treasury. It is only possible in two cases:
1. `MultiLocation` is Zeitgeists location
2. `MultiLocation` is registered in `AssetRegistry`

In all other cases (unknown tokens), fees are just burned. We can be sure that no assets are created out of thin air, because the logic that invokes the function is written by Parity and the logic that determines the fees is written and tested by us. In addition to that, we can only withdraw as much fees as have been included in an XCM transfer, that is ensured by Parities XCM implementation.